Classes

Gamb1no0 opened the pull request.

Gamb1no0 committed:
iter1

Gamb1no0 committed:
iter2

Gamb1no0 committed:
add_readme

Gamb1no0 committed:
erasing_mistake

Gamb1no0 committed:
new file:   classes_hw

Gamb1no0 committed:
modified:   README.md

Gamb1no0 committed:
modified:   README.md

Gamb1no0 committed:
modified:   classes_hw

alexandraroots commented on classes_hw:
Это не .py файл, его невозможно запустить

alexandraroots commented on classes_hw:
```diff
@@ -0,0 +1,47 @@
+        каждого слова.
+        '''
+
+        self.__feature_names = self.make_feature_names(corpus)
```
Атрибуты класса хорошо бы объявлять в методе __init__, чтобы случайно по коду не натыкаться на новые атрибуты, о существовании которых никто не подозревал

alexandraroots replied:
Так же не выйдет вызвать метод get_feature_names до fit_transform. Мы кинем в пользователя не очень понятную для него ошибку. Ожидаемое поведение в данном случае: отдавать пустой список или None, либо объяснить пользователю, почему так делать нельзя

alexandraroots commented on classes_hw:
```diff
@@ -0,0 +1,47 @@
+    def get_feature_names(self) -> List[str]:
+        '''Возвращает значение атрибута класса'''
+
+        return self.__feature_names
```
Отсутсвует метод main для запуска

alexandraroots reviewed (commented):
-2б за отсутсвие main
-3б за неверный тип файла
-1б за позднюю сдачу
-1б за некорректное поведение при вызове сначала get_features_name

3/10

Gamb1no0 committed:
conveer to py

Gamb1no0 committed:
new file:   __pycache__/countvectorizer.cpython-311.pyc
	renamed:    classes_hw -> countvectorizer.py
	new file:   main.py

Gamb1no0 commented:
Спасибо. Учел замечания, переделал.
https://github.com/Gamb1no0/omd/tree/classes

С уважением,
Мамедов Рамиль


пн, 13 нояб. 2023 г. в 21:19, Alexandra Korneeva ***@***.***>:

> ***@***.**** commented on this pull request.
>
> -2б за отсутсвие main
> -3б за неверный тип файла
> -1б за позднюю сдачу
> -1б за некорректное поведение при вызове сначала get_features_name
>
> 3/10
> ------------------------------
>
> On classes_hw
> <https://github.com/Gamb1no0/omd/pull/3#discussion_r1391495174>:
>
> Это не .py файл, его невозможно запустить
> ------------------------------
>
> In classes_hw
> <https://github.com/Gamb1no0/omd/pull/3#discussion_r1391495783>:
>
> > +                raise TypeError(f'String object expected but get {type(text)}')
> +            for word in map(lambda x: x.lower(), text.split()):
> +                if word not in set_words:
> +                    set_words.add(word)
> +                    feature_names.append(word)
> +        return feature_names
> +
> +    def fit_transform(self, corpus: List[str]) -> List[List[int]]:
> +        '''Возвращает терм-документную матрицу
> +
> +        Получает на вход коллекцию текстов и конвертирут коллекцию
> +        в набор вектор с подсчитанным колличеством вхождением
> +        каждого слова.
> +        '''
> +
> +        self.__feature_names = self.make_feature_names(corpus)
>
> Атрибуты класса хорошо бы объявлять в методе *init*, чтобы случайно по
> коду не натыкаться на новые атрибуты, о существовании которых никто не
> подозревал
> ------------------------------
>
> In classes_hw
> <https://github.com/Gamb1no0/omd/pull/3#discussion_r1391499693>:
>
> > +        каждого слова.
> +        '''
> +
> +        self.__feature_names = self.make_feature_names(corpus)
> +        count_matrix = [[0 for _ in range(len(self.__feature_names))]
> +                        for _ in range(len(corpus))]
> +        for index_text in range(len(corpus)):
> +            for word in corpus[index_text].lower().split():
> +                index_word = self.__feature_names.index(word)
> +                count_matrix[index_text][index_word] += 1
> +        return count_matrix
> +
> +    def get_feature_names(self) -> List[str]:
> +        '''Возвращает значение атрибута класса'''
> +
> +        return self.__feature_names
>
> Отсутсвует метод main для запуска
> ------------------------------
>
> In classes_hw
> <https://github.com/Gamb1no0/omd/pull/3#discussion_r1391496563>:
>
> > +                raise TypeError(f'String object expected but get {type(text)}')
> +            for word in map(lambda x: x.lower(), text.split()):
> +                if word not in set_words:
> +                    set_words.add(word)
> +                    feature_names.append(word)
> +        return feature_names
> +
> +    def fit_transform(self, corpus: List[str]) -> List[List[int]]:
> +        '''Возвращает терм-документную матрицу
> +
> +        Получает на вход коллекцию текстов и конвертирут коллекцию
> +        в набор вектор с подсчитанным колличеством вхождением
> +        каждого слова.
> +        '''
> +
> +        self.__feature_names = self.make_feature_names(corpus)
>
> Так же не выйдет вызвать метод get_feature_names до fit_transform. Мы
> кинем в пользователя не очень понятную для него ошибку. Ожидаемое поведение
> в данном случае: отдавать пустой список или None, либо объяснить
> пользователю, почему так делать нельзя
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Gamb1no0/omd/pull/3#pullrequestreview-1727931317>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU63HY23GNLENHGKKYG3D6TYEJQDNAVCNFSM6AAAAAA7AH2EKOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOMRXHEZTCMZRG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

alexandraroots commented on countvectorizer.py:
```diff
@@ -0,0 +1,50 @@
+        self.__feature_names = self.make_feature_names(corpus)
+        count_matrix = [[0 for _ in range(len(self.__feature_names))]
+                        for _ in range(len(corpus))]
+        for index_text in range(len(corpus)):
```
Везде, где есть range(len()) стоит попробовать воспользоваться enumarate

alexandraroots commented on countvectorizer.py:
```diff
@@ -0,0 +1,50 @@
+        return feature_names
+
+    def fit_transform(self, corpus: List[str]) -> List[List[int]]:
+        '''Возвращает терм-документную матрицу
```
Для докстрингов все-таки принято выбирать двойные кавычки

alexandraroots reviewed (commented):
+2б за исправленный main
+3б за исправленный  неверный тип файла
+1б за исправленный поведение при вызове сначала get_features_name

9/10

Gamb1no0 commented:
Здравствуйте, учел замечания в ревью на степике, переделал проект
<https://github.com/Gamb1no0/final_project_AAA/tree/project_dev>.

P.S. Что значит "нет PR"? [image: image.png]

С уважением,
Мамедов Рамиль

пн, 18 дек. 2023 г. в 10:32, Alexandra Korneeva ***@***.***>:

> ***@***.**** commented on this pull request.
>
> +2б за исправленный main
> +3б за исправленный неверный тип файла
> +1б за исправленный поведение при вызове сначала get_features_name
>
> 9/10
> ------------------------------
>
> In countvectorizer.py
> <https://github.com/Gamb1no0/omd/pull/3#discussion_r1429597308>:
>
> > +                    set_words.add(word)
> +                    feature_names.append(word)
> +        return feature_names
> +
> +    def fit_transform(self, corpus: List[str]) -> List[List[int]]:
> +        '''Возвращает терм-документную матрицу
> +
> +        Получает на вход коллекцию текстов и конвертирут коллекцию
> +        в набор вектор с подсчитанным колличеством вхождением
> +        каждого слова.
> +        '''
> +
> +        self.__feature_names = self.make_feature_names(corpus)
> +        count_matrix = [[0 for _ in range(len(self.__feature_names))]
> +                        for _ in range(len(corpus))]
> +        for index_text in range(len(corpus)):
>
> Везде, где есть range(len()) стоит попробовать воспользоваться enumarate
> ------------------------------
>
> In countvectorizer.py
> <https://github.com/Gamb1no0/omd/pull/3#discussion_r1429597506>:
>
> > +        set_words = set()
> +        feature_names = []
> +
> +        if not isinstance(corpus, list):
> +            raise TypeError('List corpus expected.')
> +        for text in corpus:
> +            if not isinstance(text, str):
> +                raise TypeError(f'String object expected but get {type(text)}')
> +            for word in map(lambda x: x.lower(), text.split()):
> +                if word not in set_words:
> +                    set_words.add(word)
> +                    feature_names.append(word)
> +        return feature_names
> +
> +    def fit_transform(self, corpus: List[str]) -> List[List[int]]:
> +        '''Возвращает терм-документную матрицу
>
> Для докстрингов все-таки принято выбирать двойные кавычки
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Gamb1no0/omd/pull/3#pullrequestreview-1785993927>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU63HY5WL2QCI2ALFLVPMQDYJ7WSNAVCNFSM6AAAAAA7AH2EKOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTOOBVHE4TGOJSG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
